src/object/cli_shard.c

```diff
@@ -776,7 +776,7 @@ dc_shard_update_size(struct rw_cb_args *rw_args, int fetch_rc)
 						    orw->orw_dkey_hash, iod) ||
 		    is_ec_parity_shard(rw_args->shard_args->auxi.obj_auxi->obj, orw->orw_dkey_hash,
 				       orw->orw_oid.id_shard)) {
-			if (uiod->iod_size != 0 && uiod->iod_size < sizes[i] && fetch_rc == 0) {
+			if (uiod->iod_size != 0 && uiod->iod_size < sizes[i]) {
```

Comment:
If fetch_rc is not zero, can we trust sizes[i]? Means "orwo->orw_iod_sizes.ca_arrays" may be non-initialized (or defined) under such case?

Reply:
here it will only call dc_shard_update_size() when rc is zero or -DER_REC2BIG, either case's size is server replied result.

Reply:
It is better to add some special check to prevent being abused in future.

Reply:
sure, there is a assertion "D_ASSERTF(fetch_rc == 0 || fetch_rc == -DER_REC2BIG ..." in the beginning of this func.